subsys/suit/utils/src/suit_plat_mem_util.c

```diff
@@ -84,12 +87,61 @@ uint8_t *suit_plat_mem_nvm_ptr_get(uintptr_t offset)
 	uintptr_t address;
 
 #if (DT_NODE_EXISTS(DT_NODELABEL(mram1x)))
-	address = (((offset)&0xEFFFFFFFUL) + (DT_REG_ADDR(DT_NODELABEL(mram1x)) & 0xEFFFFFFFUL));
+	address = (((offset) & 0xEFFFFFFFUL) + (DT_REG_ADDR(DT_NODELABEL(mram1x)) & 0xEFFFFFFFUL));
 #elif (DT_NODE_EXISTS(DT_NODELABEL(mram10)))
-	address = (((offset)&0xEFFFFFFFUL) + (DT_REG_ADDR(DT_NODELABEL(mram10)) & 0xEFFFFFFFUL));
+	address = (((offset) & 0xEFFFFFFFUL) + (DT_REG_ADDR(DT_NODELABEL(mram10)) & 0xEFFFFFFFUL));
 #else
 	address = offset;
 #endif
 
 	return suit_plat_mem_ptr_get(address);
 }
+
+#ifdef CONFIG_SDFW_IS_UROT
+static nrf_mramc_mode_write_t set_mramc_write_mode(nrf_mramc_mode_write_t new_write_mode)
+{
+	NRF_MRAMC_Type * const mramc = (NRF_MRAMC_Type * const)DT_REG_ADDR(DT_NODELABEL(mramc));
+	nrf_mramc_config_t mramc_config;
+
+	nrf_mramc_config_get(mramc, &mramc_config);
+
+	nrf_mramc_mode_write_t previous_write_mode = mramc_config.mode_write;
+
+	if (new_write_mode != previous_write_mode) {
+		mramc_config.mode_write = new_write_mode;
+		nrf_mramc_config_set(mramc, &mramc_config);
+	}
+
+	return previous_write_mode;
+}
+
+bool suit_plat_mem_clear_sicr_update_registers(void)
+{
+	bool cleared = true;
+
+	if (NRF_SICR->UROT.UPDATE.OPERATION != SICR_UROT_UPDATE_OPERATION_OPCODE_Nop ||
+	    NRF_SICR->UROT.UPDATE.STATUS != SICR_UROT_UPDATE_STATUS_CODE_None) {
+
+		const nrf_mramc_mode_write_t new_write_mode = NRF_MRAMC_MODE_WRITE_DIRECT;
```

Comment:
Is it really necessary to change MRAM write mode here? Shouldn't direct mode be enabled during SDFW startup?

Reply:
During development I added some logs to show current  write mode and as you say, it was already set as you describe.
However, this snippet is based on sdfw_update.c code, where such write mode is being done before changing the register values, so I assumed it is better to ensure it.
Additionally, the mode change is only done when desired mode is different than current one, so I guess it shouldn't be harmful.

Reply:
Piece of code you are proposing works as intended. At the same time - improper handling of MRAM mode may lead to data incoherency, so it shall be managed in SDFW in a coherent way, and local 'hacks' should be, in my opinion, avoided. 

Let me approve it to avoid further blocking of that PR, but please setup a separate thread to deal with MRAM mode in a proper way.

Reply:
I agree, if we can avoid additional checks and mode setting, we should remove this mechanism.
I've created a Jira task to follow it up: https://nordicsemi.atlassian.net/browse/NCSDK-30143.